### ISSUE TITLE
lint: floatCompare tuning for less false-positives

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#511]
We have many warnings on GOROOT.
Many (most) of them look like false-positives.
It's better to constrain floatCompare to be less greedy (or this should be controlled by config).

For example, comparison with zero usually makes sense, same goes for Inf and self-comparison like `x == x` or `x != x` (the latter is NaN check).

We may also want to skip simple `v != literal` and `v == literal` unless `v` is a expression, not just a var:
```go
v != 5.0 // OK
x + y != 5.0 // suspicious
foo() != 5.0 // suspicious
// and so on...
```

**Comment by Quasilyte** [comment]
CC @ludweeg
